service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCMOController.java

```diff
@@ -120,20 +110,19 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
     @PostMapping("/submitted")
     public void handleSubmittedEvent(@RequestHeader(value = "authorization") String authorization,
                                      @RequestHeader(value = "user-id") String userId,
-                                     @RequestBody CallbackRequest callbackRequest) {
-        applicationEventPublisher.publishEvent(new CMOEvent(callbackRequest, authorization, userId));
+                                     @RequestBody CallbackRequest callbackRequest) throws IOException {
+        CaseData caseData = mapper.convertValue(callbackRequest.getCaseDetails().getData(), CaseData.class);
+
+        DocmosisDocument docmosisDocument = actionCmoService.getDocmosisDocument(caseData,
```

Comment:
Story says `Once Judge has approved the CMO...`. Does the doc need to be generated if CMO is not approved?

Reply:
Why document is generated again if it was already uploaded to DM store and it was linked to the case in `/about-to-submit` callback? Why not to retrieve it from DM store instead?

Reply:
yes, the document is generated in `/about-to-submit`. I wasn't sure how dm-store handles the authentication and redirection to said document. So went with the option of generating the document (without uploading).

Reply:
https://github.com/hmcts/document-management-client/blob/master/src/main/java/uk/gov/hmcts/reform/document/DocumentDownloadClientApi.java#L16 here client that enables document download.

Reply:
Had a chat with @akomisarek , please refactor to download if time allows (preferred course of action), if not we have to accept it as debt.